main.go

```diff
@@ -33,19 +48,56 @@ func main() {
 		_ = resp.Body.Close()
 	}()
 
+	// Check that the domain specified by the user exists in GoDaddy
+	domainExists := 0
 	for _, zone := range zones {
-		if zone.Status == "ACTIVE" {
-			log.Println(zone.Domain)
+		if strings.EqualFold(zone.Status, "active") && strings.EqualFold(zone.Domain, *userDomainName) {
+			domainExists = 1
+			log.Println("Domain", zone.Domain, "matches user domain", *userDomainName)
 		}
 	}
 
-	records, _, err := apiClient.V1Api.RecordGet(ctx, "lwts.org", "", "", nil)
+	if domainExists != 1 {
+		log.Println("Domain", *userDomainName, "does not match any domains returned from the API")
+		os.Exit(1)
+	}
+
+	var recordData map[string]string
+	recordData = make(map[string]string)
+
+	records, _, err := apiClient.V1Api.RecordGet(ctx, *userDomainName, "", "", nil)
 	for _, record := range records {
-		log.Println(record.Name, "=>", record.Data)
+		if record.Name == "@" {
+			if record.Type_ == "A" {
+				log.Println("Changing", record.Type_, record.Data, "to", *userIPAddress)
+				// Fill in the recordData map so it can be used outside this loop
+				recordData["type"] = "A"
+				recordData["name"] = "@"
+
+			}
+		}
+		//log.Println(record.Name, "=>", record.Data)
 	}
 
-	domainInfo, _, err := apiClient.V1Api.Get(ctx, "lwts.org", nil)
-	log.Println(domainInfo)
+	if *applyBool {
+		changeDataArray := []godaddy.DnsRecordCreateTypeName{{
+			Data: *userIPAddress,
+			Ttl:  600,
+		}}
 
-	fmt.Print()
+		result, _ := apiClient.V1Api.RecordReplaceTypeName(ctx, *userDomainName, "A", "@", changeDataArray, nil)
+
+		if result.StatusCode == 200 {
+			log.Println("Changed", recordData["type"], recordData["name"], "to", *userIPAddress)
+			os.Exit(0)
+		}
+		if result.StatusCode != 200 {
```

Comment:
make this the first `if` statement, and use `log.Fatalln()` instead of `os.Exit()`